app/models/create_cap_proposal.rb

```diff
@@ -0,0 +1,90 @@
+class CreateCapProposal
+  def initialize(auction)
+    if !auction.is_a?(Presenter::Auction)
+      fail "an instance of Auction::Presenter must be passed to CreateCapProposal::initialize"
+    end
+
+    @auction = auction
+  end
+
+  def configure_c2_client!
+    @host = ENV.fetch('C2_HOST', 'https://cap.18f.gov')
+    @c2_client = C2::Client.new(
+      oauth_key: ENV.fetch('C2_OAUTH_KEY'),
+      oauth_secret: ENV.fetch('C2_OAUTH_SECRET'),
+      host: @host,
+      debug: ENV.fetch('C2_DEBUG', false)
+    )
+
+    @c2_client
+  end
+
+  def c2_client
+    @c2_client ||= configure_c2_client!
```

Comment:
since `configure_c2_client!` does not destroy data, I would expect it to be called `configure_c2_client` (save the bangs for destructive behaviors)